File: GithubGists/okhttplibrary/src/main/java/ru/markusfrost/okhttplibrary/ApiKeyInterceptor.java
package ru.markusfrost.okhttplibrary;

import java.io.IOException;

import androidx.annotation.NonNull;
import okhttp3.Interceptor;
import okhttp3.Response;

public final class ApiKeyInterceptor implements Interceptor {

    private ApiKeyInterceptor() {
    }

    @NonNull
    public static Interceptor create() {
        return new ApiKeyInterceptor();
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        return chain.proceed(chain.request());
    }
}

File: GithubGists/okhttplibrary/src/main/java/ru/markusfrost/okhttplibrary/ApiKeyInterceptor.kt
package ru.markusfrost.okhttplibrary

import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class ApiKeyInterceptor private constructor() : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        return chain.proceed(chain.request())
    }

    companion object {

        fun create(): Interceptor {
            return ApiKeyInterceptor()
        }
    }
}
